playground/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+}
+
+#[wasm_bindgen]
+pub fn find_captures(pattern: &str, text: &str, flags: JsValue) -> Result<JsValue, JsValue> {
+    let flags = get_flags(flags)?;
+    let regex = build_regex(pattern, &flags)?;
+
+    // Get capture group names once for this regex
+    let capture_names: Vec<Option<&str>> = regex.capture_names().collect();
+
+    let mut all_captures = Vec::new();
+
+    for caps_result in regex.captures_iter(text) {
+        match caps_result {
+            Ok(caps) => {
+                let full_match = caps.get(0).map(|m| Match {
+                    start: m.start(),
+                    end: m.end(),
+                    text: m.as_str().to_string(),
+                });
+
+                let mut captures = Vec::new();
+                for i in 0..caps.len() {
+                    let name = capture_names
+                        .get(i)
+                        .and_then(|&name_opt| name_opt.map(|s| s.to_string()));
+                    let capture = if let Some(m) = caps.get(i) {
+                        CaptureGroup {
+                            index: i,
+                            name,
+                            start: Some(m.start()),
+                            end: Some(m.end()),
+                            text: Some(m.as_str().to_string()),
+                        }
+                    } else {
+                        CaptureGroup {
+                            index: i,
+                            name,
+                            start: None,
+                            end: None,
+                            text: None,
+                        }
+                    };
+                    captures.push(capture);
+                }
+
+                all_captures.push(MatchResult {
+                    full_match,
+                    captures,
+                });
+            }
+            Err(e) => return Err(JsValue::from_str(&format!("Capture error: {}", e))),
+        }
+    }
+
+    serde_wasm_bindgen::to_value(&all_captures)
+        .map_err(|e| JsValue::from_str(&format!("Serialization error: {}", e)))
+}
+
+#[wasm_bindgen]
+pub fn parse_regex(pattern: &str, flags: JsValue) -> Result<String, JsValue> {
+    let flags = get_flags(flags)?;
+    let regex_flags = compute_regex_flags(&flags);
+
+    match fancy_regex::Expr::parse_tree_with_flags(pattern, regex_flags) {
+        Ok(tree) => Ok(format!("{:#?}", tree)),
+        Err(e) => Err(JsValue::from_str(&format!("Parse error: {}", e))),
+    }
+}
+
+#[wasm_bindgen]
+pub fn analyze_regex(pattern: &str, flags: JsValue) -> Result<String, JsValue> {
+    let flags = get_flags(flags)?;
+    let regex_flags = compute_regex_flags(&flags);
+
+    use fancy_regex::internal::{analyze, optimize};
+
+    match fancy_regex::Expr::parse_tree_with_flags(pattern, regex_flags) {
+        Ok(mut tree) => {
+            let requires_capture_group_fixup = optimize(&mut tree);
+            match analyze(&tree, if requires_capture_group_fixup { 0 } else { 1 }) {
```

Comment:
Can we put this into the crate itself? Seems a bit weird to have these internals out here.

Reply:
done